Test failure: `com.linecorp.armeria.common.multipart.MultipartDecoderTest.[1] subscriberType=INFINITE`

```
java.lang.AssertionError: 
Expecting
  <CompletableFuture[Completed: null]>
to be done.
Be aware that the state of the future in this message might not reflect the one at the time when the assertion was performed as it is evaluated later on
	at com.linecorp.armeria.common.multipart.MultipartDecoderTest.testBodyPartContentRequestMultipleTimes(MultipartDecoderTest.java:308)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

```

--- a/core/src/test/java/com/linecorp/armeria/common/multipart/MultipartDecoderTest.java
+++ b/core/src/test/java/com/linecorp/armeria/common/multipart/MultipartDecoderTest.java
@@ -152,7 +152,7 @@ public class MultipartDecoderTest {
         final AtomicLong upstreamRequestCount = new AtomicLong();
         partsPublisher(boundary, chunk1, upstreamRequestCount).subscribe(testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(headers).containsExactly("part1", "part2");
         assertThat(bodies).containsExactly("body 1", "body 2");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(1);
@@ -201,7 +201,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2), upstreamRequestCount).subscribe(
                 testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(contentIds).containsExactly("part1", "part2", "part3", "part4", "part5");
         assertThat(bodies).containsExactly("body 1", "body 2", "body 3", "body 4", "body 5");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(3);
@@ -238,7 +238,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2, chunk3, chunk4),
                        upstreamRequestCount).subscribe(testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(contentIds).containsExactly("part1");
         assertThat(bodies).containsExactly(
                 "this-is-the-1st-slice-of-the-body\n" +
@@ -305,7 +305,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2, chunk3, chunk4),
                        upstreamRequestCount).subscribe(testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(contentIds).containsExactly("part1");
         assertThat(bodies).containsExactly(
                 "this-is-the-1st-slice-o",
@@ -368,7 +368,7 @@ public class MultipartDecoderTest {
                        upstreamRequestCount)
                 .subscribe(testSubscriber);
         await().forever().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(bodies).containsExactly("body 1", "body 2");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(11);
     }
@@ -406,7 +406,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2, chunk3, chunk4, chunk5), upstreamRequestCount)
                 .subscribe(testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(headers).containsExactly("part1", "text/plain", "bob=alice", "foo=bar");
         assertThat(bodies).containsExactly("body 1");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(6);
@@ -467,7 +467,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2, chunk3, chunk4, chunk5), upstreamRequestCount)
                 .subscribe(testSubscriber);
         await().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(headers).containsExactly("part1", "text/plain", "bob=alice", "foo=bar");
         assertThat(bodies).containsExactly("body 1");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(6);
@@ -512,7 +512,7 @@ public class MultipartDecoderTest {
         final AtomicLong upstreamRequestCount = new AtomicLong();
         partsPublisher(boundary, chunk1, upstreamRequestCount).subscribe(testSubscriber);
         await().forever().untilAtomic(counter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(bodies).containsExactly("body 1", "body 2");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(2);
     }
@@ -627,7 +627,7 @@ public class MultipartDecoderTest {
         partsPublisher(boundary, ImmutableList.of(chunk1, chunk2, chunk3, chunk4),
                        upstreamRequestCount).subscribe(testSubscriber);
         await().untilAtomic(counter, is(1));
-        assertThat(testSubscriber.completionFuture).isNotDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isNotDone());
         assertThat(headers).containsExactly("part1");
         assertThat(upstreamRequestCount.get()).isLessThanOrEqualTo(2);
     }
@@ -820,7 +820,7 @@ public class MultipartDecoderTest {
                        upstreamRequestCount)
                 .subscribe(testSubscriber);
         await().untilAtomic(completeCounter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(headers).containsExactly("part1", "part2", "part3", "part4");
         assertThat(bodies).containsExactly("this-is-the-1st-slice-o", "body 2",
                                            "this-is-the-1st-slice-of", "body 4");
@@ -884,7 +884,7 @@ public class MultipartDecoderTest {
                        upstreamRequestCount)
                 .subscribe(testSubscriber);
         await().untilAtomic(completeCounter, is(0));
-        assertThat(testSubscriber.completionFuture).isDone();
+        await().untilAsserted(() -> assertThat(testSubscriber.completionFuture).isDone());
         assertThat(headers).containsExactly("part1");
         assertThat(bodies).containsExactly("this-is-the-1st-slice-o",
                                            "f-the-body\nthis-is-the-2nd-slice-o",

